packages/@react-aria/focus/test/FocusScope.test.js

```diff
@@ -1251,4 +1251,46 @@ describe('FocusScope', function () {
       expect(document.activeElement).toBe(beforeScope);
     });
   });
+
+  describe('focusableFirstInScope', function () {
+    it('should restore focus to the first focusable or tabbable element within the scope when focus is lost within the scope', function () {
+      let {getByTestId} = render(
+        <div>
+          <FocusScope contain>
+            <div role="dialog" data-testid="focusable" tabIndex={-1}>
+              <button data-testid="tabbable1" onClick={e => act(() => e.target.remove())}>Remove me!</button>
+              <Item data-testid="item1" tabIndex={0}>Remove me, too!</Item>
+              <Item data-testid="item2" tabIndex={-1}>Remove me, three!</Item>
+            </div>
+          </FocusScope>
+        </div>
+      );
+
+      function Item(props) {
+        let focusManager = useFocusManager();
+        let onClick = e => {
+          focusManager.focusNext();
+          act(() => e.target.remove());
+        };
+        return <button tabIndex={-1} {...props} onClick={onClick} />;
+      }
+      let focusable = getByTestId('focusable');
+      let tabbable1 = getByTestId('tabbable1');
+      let item1 = getByTestId('item1');
+      let item2 = getByTestId('item2');
+      act(() => tabbable1.focus());
+      expect(document.activeElement).toBe(tabbable1);
+      fireEvent.click(tabbable1);
+      expect(tabbable1).not.toBeInTheDocument();
+      waitFor(() => expect(document.activeElement).toBe(focusable));
```

Comment:
So I had expected this to fail since the FocusScope logic should have moved focus to "item1" instead of to the dialog since it prioritizes tabbable elements over focusable ones. Did some digging, reproduced the expected behavior in the browser but the test doesn't seem to fire a blur even when the button is removed on click.

Tried to resolve it locally by making the requestAnimationFrame mock async with a setTimeout + jest.fakeTimers and manually firing a blur before the e.target.remove but it breaks the test with `Cannot read property 'createEvent' of null` error on react dom. Will need to dig more. 

If we can't get the test to mimic the browser behavior, I feel like we should drop this test or rewrite it (if possible, dunno if the same error is gonna happen)

Reply:
@LFDanLu I think I have managed to fix the test with https://github.com/adobe/react-spectrum/pull/2763/commits/a6e485fc4fd2fae38bbd8e1e159e287f678bf943.